synapse/metrics/_exposition.py

```diff
@@ -39,15 +39,11 @@
 CONTENT_TYPE_LATEST = "text/plain; version=0.0.4; charset=utf-8"
 
 
-INF = float("inf")
-MINUS_INF = float("-inf")
-
-
-def floatToGoString(d: Any) -> str:
+def floatToGoString(d: Union[int, float]) -> str:
```

Comment:
```suggestion
def floatToGoString(d: float) -> str:
```

I think `int` is a subtype of `float`

Reply:
hm, actually, I'm pretty minded to leave it as is to signal that we explicitly thought about both ints and floats.

Python ints are arbitrary-sized, so it'd feel funny to have it accept float and then still have to call `float(d)` (which is understandable because presumably it will cut off very big ints as 'infinity'?)

Reply:
Never mind:

```python
>>> float(2 ** 3000)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: int too large to convert to float
```

Reply:
Normally I'd go for `float` over `Union[int, float]`, but then the `float()` cast looks silly.  Without the cast, the function changes behaviour and omits the decimal point for integer values (making some tests fail).

Reply:
Ah I see — in that case, LGTM leaving it as a union